nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-resources/src/main/resources/conf/bootstrap.conf

```diff
@@ -45,75 +42,14 @@ java.arg.3=-Xmx${nifi.jvm.heap.max}
 java.arg.5=-Dsun.net.http.allowRestrictedHeaders=true
 java.arg.6=-Djava.protocol.handler.pkgs=sun.net.www.protocol
 
-#Set headless mode by default
-java.arg.14=-Djava.awt.headless=true
+# Enable Headless mode to avoid HeadlessException with Java AWT libraries
+java.arg.headless=-Djava.awt.headless=true
+
+# Configure Apache Curator connection logging for Apache ZooKeeper to avoid excessive ERROR messages
+java.arg.curatorLogOnlyFirstConnectionIssue=-Dcurator-log-only-first-connection-issue-as-error-level=true
 
 # Root key in hexadecimal format for encrypted sensitive configuration values
 nifi.bootstrap.sensitive.key=
 
-# Sensitive Property Provider configuration
-
-# HashiCorp Vault Sensitive Property Providers
-#nifi.bootstrap.protection.hashicorp.vault.conf=./conf/bootstrap-hashicorp-vault.conf
-
-# AWS Sensitive Property Providers
-#nifi.bootstrap.protection.aws.conf=./conf/bootstrap-aws.conf
-
-# Azure Key Vault Sensitive Property Providers
-#nifi.bootstrap.protection.azure.keyvault.conf=./conf/bootstrap-azure.conf
-
-# GCP KMS Sensitive Property Providers
-#nifi.bootstrap.protection.gcp.kms.conf=./conf/bootstrap-gcp.conf
-
-# Sets the provider of SecureRandom to /dev/urandom to prevent blocking on VMs
-java.arg.15=-Djava.security.egd=file:/dev/urandom
-
-# Requires JAAS to use only the provided JAAS configuration to authenticate a Subject, without using any "fallback" methods (such as prompting for username/password)
-# Please see https://docs.oracle.com/en/java/javase/17/security/single-sign-using-kerberos-java1.html, section "EXCEPTIONS TO THE MODEL"
-java.arg.16=-Djavax.security.auth.useSubjectCredsOnly=true
```

Comment:
@exceptionfactory I am not aware of any situation in NiFi usage whereby we'd ever want someone with a Kerberos setup to allow any prompting for user input at the command line.  This prevents that.  I recommend we retain this.

Reply:
That's a good point, it is common enough that it is a useful default. I reverted the changes.